### PR TITLE
Make sparse([1,1], [1,1], [true,true]) work

### PR DESCRIPTION
**Pull request by andreasnoack**
...by making sure that the sparse method for Bool is always used.

As reported in https://groups.google.com/forum/#!topic/julia-dev/QT7qpIpgOaA

### CODE REVIEW

**Comment by tkelman** [comment]
Huh, I didn't realize we had this special case from L333: `sparse(I,J,V::AbstractVector{Bool},m,n) = sparse(I, J, V, Int(m), Int(n), OrFun())`. We should document that the default combining behavior is or-ing rather than adding for arrays of booleans.

**Comment by andreasnoack** [comment]
Neither did I, but it is actually quite old https://github.com/JuliaLang/julia/commit/bc8fdb1b77876428ad23b9d006ca48f55b30148a. I'll update the documentation here as well.

**Comment by andreasnoack** [comment]
Docs updated. Ready for review.

File: base/sparse/csparse.jl [hunk 1]
```diff
@@ -19,8 +19,9 @@
 Create a sparse matrix `S` of dimensions `m x n` such that `S[I[k], J[k]] = V[k]`.
 The `combine` function is used to combine duplicates. If `m` and `n` are not
 specified, they are set to `maximum(I)` and `maximum(J)` respectively. If the
-`combine` function is not supplied, duplicates are added by default. All elements
-of `I` must satisfy `1 <= I[k] <= m`, and all elements of `J` must satisfy `1 <= J[k] <= n`.
+`combine` function is not supplied, `combine` defaults to addition except for
+Booleans for which `combine` defaults to disjunction. All elements of `I` must satisfy
```
Comment (tkelman) [inline base/sparse/csparse.jl]: "except for Booleans" could be clearer that it's referring to tge element type of the nonzeros input.

disjunction is too jargon-y, a simpler description would be better

File: base/sparse/csparse.jl [hunk 2]
```diff
@@ -19,8 +19,9 @@
 Create a sparse matrix `S` of dimensions `m x n` such that `S[I[k], J[k]] = V[k]`.
 The `combine` function is used to combine duplicates. If `m` and `n` are not
 specified, they are set to `maximum(I)` and `maximum(J)` respectively. If the
-`combine` function is not supplied, duplicates are added by default. All elements
-of `I` must satisfy `1 <= I[k] <= m`, and all elements of `J` must satisfy `1 <= J[k] <= n`.
+`combine` function is not supplied, `combine` defaults to `+` except for Boolean
```
Comment (tkelman) [inline base/sparse/csparse.jl]: Should spell out that elements of `V` are what matters. "`combine` defaults to `+`, unless the elements of `V` are Boolean in which case `combine` defaults to `|`" ?

File: base/sparse/sparsevector.jl [hunk 3]
```diff
@@ -122,7 +122,8 @@ end
 
 Create a sparse vector `S` of length `m` such that `S[I[k]] = V[k]`.
 Duplicates are combined using the `combine` function, which defaults to
-`+` if it is not provided.
+`+` if no combine argument is provided except for Boolean elements
```
Comment (tkelman) [inline base/sparse/sparsevector.jl]: "if no `combine` argument is provided, unless the elements of `V` are Boolean in which case combine defaults to `|`" ?

**Comment by andreasnoack** [comment]
Updated

**Pull request merged**